### DISCUSSION TITLE
How to debug container 'unhealthy'

### DISCUSSION THREAD

**Discussion by dakerman** [mbround18/valheim-docker#660] [Help!]
Hi! I just started out hosting Valheim for the first time a few weeks back. I don't have any mods and it's pretty much vanilla configuration. The first couple of weeks everything was working perfectly but the last week or so the container ends up unhealthy and it's not possible to logon in the game.

What I've noticed
* Restarting container always resolves the issue.
* Looking in the  valheim_server.err I only have one entry which seems unrelated. I'll post the full log below.
* (update: NOT TRUE) It never happens when playing, only when idle. 

Is there any other logs I should be looking at? Or any other steps to debug this further? Thank you!

`[S_API] SteamAPI_Init(): Loaded local 'steamclient.so' OK.
CAppInfoCacheReadFromDiskThread took 4 milliseconds to initialize
Setting breakpad minidump AppID = 892970
[S_API FAIL] Tried to access Steam interface SteamNetworkingUtils003 before SteamAPI_Init succeeded.`

Update: I've added some more monitoring now and I just got kicked out while playing. 

Total CPU usage is steady between 7~14%
Memory is around 75% for 10-15 minutes then it slowly increases up to around 89%. Could this be a memory issue? It seems like it's never hitting 100% and I got no swap at all.

**Comment by Bmiest** [comment]
I have been having the same issue, worked great for, let's say, almost more than a year. 
But recently none of my containers seem to be working anymore: 
![afbeelding](https://user-images.githubusercontent.com/18619887/218096781-5d264e12-2a3d-48a1-8044-97635a76a9a3.png)
They are all unhealthy.

There's nothing really in the logs telling me why

**Reply by mbround18** [reply]
@Bmiest it looks like those containers are failing their healthcheck https://github.com/mbround18/valheim-docker/blob/main/Dockerfile.valheim#LL93-L94C46

Perhaps, you could try and restart them? If you have tried that, run 

```sh
docker exec valheim2_valheim_1 gosu steam bash
```

This will get you into the steam profile on the container, then run:

```sh
htop
```

in the container, is the valheim server actually running? if not, all logs are stored in `/home/steam/valheim/logs`, if you grab those do they contain any errors as to why the server might be crashing?

**Comment by mbround18** [comment]
@dakerman  apologies i was moving cross country, the rust code in this project is just a wrapper to launch the application & then all memory usage from that point on is based on the valheim server itself. 

@Bmiest I can look into why this might be happening, are you still able to access your servers?

**Reply by dakerman** [reply]
No worries @mbround18, thanks for the reply. Seems like @Bmiest and me have the same issue. After a restart the container is healthy for about 10 minutes before reporting unhealthy again.

I did just bump the machine for now, running on 8Gb instead of 4 and have not seen the issue since. So it seems to be oom even though I never see the memory at 100% or any swap.

**Reply by mbround18** [reply]
@dakerman that is what I've found too, the game tends to be a bit chunky and 4gb memory is the minimum

**Comment by spannerman79** [comment]
>Memory is around 75% for 10-15 minutes then it slowly increases up to around 89%. Could this be a memory issue? It seems like it's never hitting 100% and I got no swap at all.

Using one of the more recent *buntu's?  If so - host issue.  *buntu has systemd-oom enabled **by default** which is pretty aggresive.

**Reply by dakerman** [reply]
This might very well be what kills it. I'll try disabling it and see if it makes any difference (running 22.04)
